File: pset1/cash/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    do
    {
        dollars = get_float("Cash owed: ");
    }
    while (dollars <= 0);

    int cents = round(dollars * 100);

    int coins = 0;

//25c, 10c, 5c, 1c

    while (cents >= 25)
    {
        cents -= 25;
        coins++;
    }

    while (cents >= 10 && cents < 25)
    {
        cents -= 10;
        coins++;
    }

    while (cents >= 5 && cents < 25 && cents < 10)
    {
        cents -= 5;
        coins++;
    }
    while (cents >= 1 && cents < 25 && cents < 10 && cents < 5)
    {
        cents -= 1;
        coins++;
    }

    printf("%i\n", coins);
}

File: pset6/cash/cash.py
while True:
    try:
        dollars = float(input("Cash owed: "))

        if dollars > 0:
            cents = dollars * 100
            coins = 0

            while cents >= 25:
                cents -= 25
                coins += 1

            while 10 <= cents < 25:
                cents -= 10
                coins += 1

            while 5 <= cents < 25 and cents < 10:
                cents -= 5
                coins += 1

            while 1 <= cents < 25 and cents < 10 and cents < 5:
                cents -= 1
                coins += 1

            print(coins)

            break

        else:
            continue

    except ValueError:
        continue